File: Leetcode/Python/_56.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort()
        merge = [intervals[0]]
        for inter in intervals[1:]:
            if inter[0] <= merge[-1][1] and inter[1] <= merge[-1][1]:
                continue
            elif inter[0] <= merge[-1][1]:
                merge[-1] = [merge[-1][0], inter[1]]
            else:
                merge.append(inter)
        return merge

File: Leetcode/cpp/56.cpp
class Solution {
public:
    std::vector<std::vector<int>> merge(std::vector<std::vector<int>>& intervals) {
        if (intervals.empty()) return {};

        std::sort(
            intervals.begin(),
            intervals.end(),
            [](const std::vector<int>& a, const std::vector<int>& b) {
                if (a[0] < b[0]) {
                    return true;
                } else if (a[0] == b[0] && a[1] > b[1]) {
                    return true;
                }
                return false;
            }
        );
        std::vector<std::vector<int>> output = {intervals[0]};
        for (int i = 1; i < intervals.size(); ++i) {
            if (output.back()[1] >= intervals[i][0]) {
                output.back()[1] = std::max<int>(output.back()[1], intervals[i][1]);
            } else {
                output.push_back(intervals[i]);
            }
        }

        return output;
    }
};
